sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/b2c/ClientRegistrationCondition.java

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.spring.autoconfigure.b2c;
+
+import org.springframework.boot.autoconfigure.condition.ConditionOutcome;
+import org.springframework.boot.autoconfigure.condition.SpringBootCondition;
+import org.springframework.boot.context.properties.bind.Binder;
+import org.springframework.context.annotation.ConditionContext;
+import org.springframework.core.type.AnnotatedTypeMetadata;
+import org.springframework.util.CollectionUtils;
+
+/**
+ * Configuration condition for activating OAuth2 client related beans.
+ */
+public class ClientRegistrationCondition extends SpringBootCondition {
```

Comment:
How many instances of this condition will be created if we put the condition on multiple beans?